.woodpecker/docs.yml

```diff
@@ -2,39 +2,40 @@ clone:
   git:
     image: plugins/git:next
 
-branches: master
-
 pipeline:
-  deploy-docs:
+  build:
     image: node:14-alpine
+    commands:
+      - cd docs/
+      - yarn install --frozen-lockfile
+      - yarn build
+    when:
+      event: [push, pull_request]
+      path: "docs/**"
+
+  deploy:
+    image: alpine:3.14.2
     secrets:
       - BOT_PRIVATE_KEY
-    environment:
-      - USE_SSH=true
-      - GIT_USER=woodpecker-bot
-      - DEPLOYMENT_BRANCH=master
     commands:
-      - apk add openssh-client git
+      - apk add openssh-client git rsync
       - mkdir -p $HOME/.ssh
       - ssh-keyscan -t rsa github.com >> $HOME/.ssh/known_hosts
       - echo "$BOT_PRIVATE_KEY" > $HOME/.ssh/id_rsa
       - chmod 0600 $HOME/.ssh/id_rsa
       - git config --global user.email "woodpecker-bot@obermui.de"
       - git config --global user.name "woodpecker-bot"
-      - cd docs/
-      - yarn install --frozen-lockfile
-      - yarn deploy
+      - git clone -b master --single-branch git@github.com:woodpecker-ci/woodpecker-ci.github.io.git /repo
+      # copy all docs files and delete all old ones, but leave CNAME and index.yaml untouched
+      - rsync -r --exclude .git --exclude CNAME --exclude index.yaml --delete docs/build/ /repo
```

Comment:
```suggestion
      - rsync -r --exclude .git --exclude README.md --exclude index.yaml --delete docs/build/ /repo
```

Reply:
what is CNAME for?

Reply:
CNAME is the CNAME file you created for the domain for the github-page.